tests/typed_mock.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
```

Comment:
This is a good candidate to have a module docstring. I feel that we understand it after your explanation, but this may look obscure for someone unfamiliar with the concept. Either that or/and a docstring to the function `create_typed_mock`